### DISCUSSION TITLE
66. InnoDB 스토리지 엔진의 잠금 중 갭 락과 넥스트 키 락에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#68] [General]
InnoDB 스토리지 엔진의 잠금 중 갭 락과 넥스트 키 락에 대해 설명해 주세요.

**Comment by eun-byeol** [comment]
갭 락(Gap Lock)은 `레코드 사이의 간격`에 잠금을 거는 것입니다. 넥스트 키락(Next-Key Lock)은 `레코드 락`과 `갭 락`이 결합된 잠금입니다. 해당 레코드 자체와 인접한 간격을 잠가 다른 트랜잭션이 해당 레코드를 삽입하거나 수정하지 못하도록 한다.

`InnoDB` 엔진에서는 기본 격리 수준(REPEATABLE READ)에서 갭 락과 넥스트 키 락을 통해 `팬텀 리드`를 방지할 수 있습니다.
